wallet/src/account/currency_grouper/mod.rs

```diff
@@ -128,6 +130,36 @@ pub fn group_outputs_with_issuance_fee<T, Grouped: Clone>(
     Ok(tokens_grouped)
 }
 
+pub fn output_currency_value(output: &TxOutput) -> Result<(Currency, Amount), UtxoSelectorError> {
```

Comment:
These kinds of functions are just pure play with fire... and I have consistently complained about them. Please avoid doing generic logical groupings like this for enums. 

A while back we had a `value()` function for TxOutput, and it was removed for the same reason I call this playing with fire: People tend to casually just call the "value" function when they want the value, while what they need in reality is a separate `match` for every different case. There's nothing such as "value" for a transaction output. Every case means a different thing. There may be a case where you need spendable values. There may be another case where you need "spendable values at any time in the future". There may be a case where you need "values locked in a pool". These umbrella functions just lead to issues and oversight by removing the protection the compiler provides. I'm more than happy to "bloat" the code with as many match statements as necessary, and in return avoid mistakes.

The only case where such a function is a good idea, is when it's private for a specific case that's included in the name of the function.